crates/mysten-network/src/multiaddr.rs

```diff
@@ -42,6 +42,14 @@ impl Multiaddr {
         self.0.replace(at, by).map(Self)
     }
 
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
```

Comment:
I don't think this is correct, in that internally what is stored is a compact format but what we have on chain is a String, so for the checks below we really need to do `.to_string()` to check them against the length.

Reply:
This is a rust tooling side problem so we can fix it later. Landing to get this in for the cut